### PR TITLE
Issue #3264287 by robertragas, Ressinel: Fix upgrading from Open Social 10 to 11 errors.

### PR DESCRIPTION
**Pull request by robertragas**
### 1) Message notify module error.
Upgrading from Open Social 10 to 11 requires pre-enabling message notify module. This is a known issue at Private message, and we need to review, and change Open Social to match their patch.

https://www.drupal.org/project/private_message/issues/3029461

**Update:** Issue https://www.drupal.org/project/private_message/issues/3029461 was fixed and released in [3.0.0-beta2](https://www.drupal.org/project/private_message/releases/3.0.0-beta2).

### 2) Hook update `views_bulk_operations_update_8034` error.
```
>  [notice] Update started: views_bulk_operations_update_8034
>  [error]  The "VariationCache" module is not installed. Please run the update script to install it properly. 
>  [error]  Update failed: views_bulk_operations_update_8034 
```

## Solution
1. Review this patch, https://www.drupal.org/project/private_message/issues/3029461, provide input or fix if possible.
**Update:** Update `private_message` module to [3.0.0-beta2](https://www.drupal.org/project/private_message/releases/3.0.0-beta2) version.
2. Convert the configuration of existing views to the new schema only after **VariationCache** is installed.

## Issue tracker
- https://www.drupal.org/project/social/issues/3264287
- https://www.drupal.org/project/private_message/issues/3029461

## How to test
- [ ] Upgrade from **Open Social 10** to **Open Social 11**. It will break if we don't have message notify enabled.
- [ ] Apply current PR changes.
- [ ] You should not have the error anymore for the first step.

## Screenshots
N/A

## Release notes
The drupal/private_message module has been updated to 3.0.0-beta2. Please view the release notes for [2.0-beta18](https://www.drupal.org/project/private_message/releases/8.x-2.0-beta18), [3.0.0-beta1](https://www.drupal.org/project/private_message/releases/3.0.0-beta1), and [3.0.0-beta2](https://www.drupal.org/project/private_message/releases/3.0.0-beta2) of that module for the changes.

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2756-auzt9vllaakhkmtjopielj4hehpsctjj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6202919200c7e8a7588eb8e5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2756-auzt9vllaakhkmtjopielj4hehpsctjj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6202919200c7e8a7588eb8e5

**Comment by robertragas** [comment]
@navneet0693 @tbsiqueira so this is the PR to fix the service not being found, but still needs a bit work, which I don't have time for tomorrow, but will check later.

So we use the patch which turns the private message notifier into a submodule, but in the submodule they actually change the method to contain 2 arguments instead of 3, and the one they removed $members is the one we actually alter. 

This has been moved to an alter hook, so think we don't actually need the service anymore but need to move it to the alter hook, that is, if we still want to use the patch.

File: composer.json [hunk 1]
```diff
@@ -116,7 +116,8 @@
                 "Integrity constraint violation: 1048 Column 'langcode' cannot be null": "https://www.drupal.org/files/issues/2019-07-10/paragraphs-set_langcode_widgets-2901390-29.patch"
             },
             "drupal/private_message": {
-                "3248258 - Coding standard issue at PrivateMessageService": "https://www.drupal.org/files/issues/2021-11-08/coding-standard-3248258-2.patch"
+                "3248258 - Coding standard issue at PrivateMessageService": "https://www.drupal.org/files/issues/2021-11-08/coding-standard-3248258-2.patch",
+                "Move message notify to a submodule so it doesn't crashes on update": "https://git.drupalcode.org/project/private_message/-/merge_requests/31.diff"
```
Comment (tbsiqueira) [inline composer.json:120]: Hi Robert, Ronald made a comment to a [similar situation here](https://github.com/goalgorilla/open_social/pull/2748/files#r802637677) we should follow the same approach :D

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2756-auzt9vllaakhkmtjopielj4hehpsctjj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6202919200c7e8a7588eb8e5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2756-auzt9vllaakhkmtjopielj4hehpsctjj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6202919200c7e8a7588eb8e5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2756-auzt9vllaakhkmtjopielj4hehpsctjj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6202919200c7e8a7588eb8e5

**Review by navneet0693 — APPROVED** [review]

**Review by navneet0693 — CHANGES_REQUESTED** [review]
```

>  [notice] Update started: views_bulk_operations_update_8034
>  [notice] 1 view configs converted.
>  [notice] Update completed: views_bulk_operations_update_8034
>  [notice] Update started: social_user_export_update_11001
>  [notice] Update completed: social_user_export_update_11001
>  [notice] Update started: social_group_update_11001
> [info] Configuration views.view.groups has been successfully updated.
>  [notice] Update completed: social_group_update_11001
>  [notice] Update started: social_group_update_11002
>  [notice] Update completed: social_group_update_11002
>  [notice] Update started: social_group_update_11003
>  [notice] Update completed: social_group_update_11003
>  [notice] Update started: social_event_managers_update_11001
>  [notice] Update completed: social_event_managers_update_11001
>  [notice] Update started: social_event_enrolments_export_update_11001
>  [notice] Update completed: social_event_enrolments_export_update_11001
>  [notice] Update started: private_message_update_8007
>  [error]  Configuration objects (message.template.private_message_notification) provided by private_message_notify already exist in active configuration
>  [error]  Update failed: private_message_update_8007
>  [notice] Update started: social_event_an_enroll_update_11001
>  [notice] Update completed: social_event_an_enroll_update_11001
 [error]  Update aborted by: private_message_update_8007
 [error]  Finished performing updates.

```

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2756-auzt9vllaakhkmtjopielj4hehpsctjj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6202919200c7e8a7588eb8e5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2756-auzt9vllaakhkmtjopielj4hehpsctjj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6202919200c7e8a7588eb8e5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2756-auzt9vllaakhkmtjopielj4hehpsctjj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6202919200c7e8a7588eb8e5

**Comment by navneet0693** [comment]
@Ressinel https://www.drupal.org/project/private_message/issues/3029461 is now merged in 3.0.x branch.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2756-auzt9vllaakhkmtjopielj4hehpsctjj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6202919200c7e8a7588eb8e5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2756-auzt9vllaakhkmtjopielj4hehpsctjj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6202919200c7e8a7588eb8e5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2756-auzt9vllaakhkmtjopielj4hehpsctjj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6202919200c7e8a7588eb8e5

**Review by tbsiqueira — APPROVED** [review]

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2756-auzt9vllaakhkmtjopielj4hehpsctjj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6202919200c7e8a7588eb8e5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2756-auzt9vllaakhkmtjopielj4hehpsctjj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6202919200c7e8a7588eb8e5

**Review by navneet0693 — APPROVED** [review]

**Comment by navneet0693** [comment]
🍒 - picked to 11.0.x and 11.1.x

**Pull request merged**